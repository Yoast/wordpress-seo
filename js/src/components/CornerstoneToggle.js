import { Component } from "@wordpress/element";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Toggle } from "@yoast/components";
import { __ } from "@wordpress/i18n";

const Cornerstone = styled.div`
	display: flex;
	margin-top: 8px;
`;

/**
 * The CornerstoneToggle Component.
 */
class CornerstoneToggle extends Component {
	/**
	 * Renders the CornerstoneToggle component.
	 *
	 * @returns {wp.Element} the CornerstoneToggle component.
	 */
	render() {
		return (
			<Cornerstone>
				<Toggle
					id={ this.props.id }
					labelText={ __( "Mark as cornerstone content", "wordpress-seo" ) }
					isEnabled={ this.props.isEnabled }
					onSetToggleState={ this.props.onToggle }
					onToggleDisabled={ this.props.onToggleDisabled }
				/>
			</Cornerstone>
		);
	}
}

CornerstoneToggle.propTypes = {
	id: PropTypes.string,
	isEnabled: PropTypes.bool,
	onToggle: PropTypes.func,
	onToggleDisabled: PropTypes.func,
};

CornerstoneToggle.defaultProps = {
	id: "cornerstone-toggle",
	isEnabled: true,
	onToggle: () => { },
	onToggleDisabled: () => { },
};

export default CornerstoneToggle;
