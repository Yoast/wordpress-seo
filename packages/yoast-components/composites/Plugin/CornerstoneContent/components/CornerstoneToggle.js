import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Toggle } from "@yoast/components";
import { __ } from "@wordpress/i18n";

const Cornerstone = styled.div`
	display: flex;
	margin-top: 8px;
`;

/**
 * The CornerstoneToggle Component.
 */
class CornerstoneToggle extends React.Component {
	/**
	 * Renders the CornerstoneToggle component.
	 *
	 * @returns {ReactElement} the CornerstoneToggle component.
	 */
	render() {
		return (
			<Cornerstone>
				<Toggle
					id={ this.props.id }
					labelText={ __( "Mark as cornerstone content", "wordpress-seo" ) }
					isEnabled={ this.props.isEnabled }
					onSetToggleState={ this.props.onToggle }
					onToggleDisabled={ this.props.onToggleDisabled }
				/>
			</Cornerstone>
		);
	}
}

CornerstoneToggle.propTypes = {
	id: PropTypes.string,
	isEnabled: PropTypes.bool,
	onSetToggleState: PropTypes.func,
	onToggle: PropTypes.func,
	disable: PropTypes.bool,
	onToggleDisabled: PropTypes.func,
};

CornerstoneToggle.defaultProps = {
	id: "cornerstone-toggle",
};

export default CornerstoneToggle;
