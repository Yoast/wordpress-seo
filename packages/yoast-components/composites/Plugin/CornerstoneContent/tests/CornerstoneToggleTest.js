import React from "react";
import renderer from "react-test-renderer";
import CornerstoneToggle from "../components/CornerstoneToggle";

test( "The CornerstoneToggle matches the snapshot", () => {
	const component = renderer.create(
		<CornerstoneToggle onChange={ () => {} } checked={ true } />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
