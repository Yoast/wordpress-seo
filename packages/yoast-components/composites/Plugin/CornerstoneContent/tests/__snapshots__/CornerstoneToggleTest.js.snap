// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CornerstoneToggle matches the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c2 {
  margin-right: 16px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

.c3 {
  background-color: #ccc;
  border-radius: 7px;
  height: 14px;
  width: 30px;
  cursor: pointer;
  margin: 0;
  outline: 0;
}

.c3:focus > span {
  box-shadow: inset 0 0 0 1px #fff,0 0 0 1px #5b9dd9,0 0 2px 1px rgba(30,140,190,.8);
}

.c4 {
  background-color: #888;
  margin-left: -2px;
  box-shadow: 0 2px 2px 2px rgba(0,0,0,0.1);
  border-radius: 100%;
  height: 20px;
  width: 20px;
  position: absolute;
  margin-top: -3px;
}

.c5 {
  font-size: 14px;
  line-height: 20px;
  margin-left: 8px;
  font-style: italic;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
      id="cornerstone-toggle"
      onClick={[Function]}
    >
      Mark as cornerstone content
    </span>
    <div
      aria-checked={false}
      aria-disabled={false}
      aria-labelledby="cornerstone-toggle"
      className="c3"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="checkbox"
      tabIndex="0"
    >
      <span
        className="c4"
      />
    </div>
    <span
      aria-hidden="true"
      className="c5"
    >
      Off
    </span>
  </div>
</div>
`;
